client.go

```diff
@@ -428,6 +446,38 @@ func (c *Client) handleMessage(msgType int, data []byte) error {
 				return err
 			}
 			c.sigChainBlockHash = sigChainBlockHash
+
+		// client auth
+		case authChallengeAction:
+			var challenge string
+			if err := json.Unmarshal(*msg["Challenge"], &challenge); err != nil {
+				return err
+			}
+			byteChallenge, err := hex.DecodeString(challenge)
+			if err != nil {
+				return err
+			}
+
+			clientSalt := make([]byte, 32)
+			_, err = rand.Reader.Read(clientSalt)
+			if err != nil {
+				return err
+			}
+
+			// add client salt to the node's challenge
+			byteChallenge = append(byteChallenge, clientSalt...)
+			hash := sha256.Sum256(byteChallenge)
+			signature, err := crypto.Sign(c.account.PrivateKey, hash[:])
+			if err != nil {
+				return err
+			}
+
+			saltAndSig := stSaltAndSignature{ClientSalt: clientSalt[:], Signature: signature}
```

Comment:
`clientSalt[:]` is the same as `clientSalt`

Reply:
Yes, modified.